clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -3564,23 +3543,188 @@ static void handleSYCLIntelSchedulerTargetFmaxMhzAttr(Sema &S, Decl *D,
 }
 
 // Handles max_global_work_dim.
-static void handleMaxGlobalWorkDimAttr(Sema &S, Decl *D, const ParsedAttr &A) {
-  if (D->isInvalidDecl())
-    return;
+// Returns a ArgCheck value; EqualToOne means all argument values are one,
+// NotEqualToOne means all argument values are not one, and Unknown means that
+// the argument values are invalid.
+enum class ArgCheck { Unknown, EqualToOne, NotEqualToOne };
+static ArgCheck AreArgValuesToOne(const Expr *E, const Expr *E1,
```

Comment:
This function is only used in one place -- should it be a lambda instead (along with the enumeration)?